### PR TITLE
[-Wsign-compare] warning fixes batch 6

### PR DESCRIPTION
**Pull request by tg-at-google**
@mihaimaruseac

### CODE REVIEW

File: tensorflow/core/framework/shape_inference.cc [hunk 1]
```diff
@@ -181,7 +182,8 @@ void InferenceContext::PreInputInit(
 }
 
 Status InferenceContext::ExpandOutputs(int new_output_size) {
-  if (new_output_size < outputs_.size()) {
+  int outputs_size_ = outputs_.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/shape_inference.cc:185]: Should be `const`

Comment (mihaimaruseac) [reply tensorflow/core/framework/shape_inference.cc:185]: Also, should not end with `_` as names ending in `_` are reserved for private class members in the coding style

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/core/framework/shape_inference.cc [hunk 2]
```diff
@@ -718,7 +720,8 @@ Status InferenceContext::MakeShapeFromShapeTensorTreatScalarAsUnknownShape(
   TF_RETURN_IF_ERROR(WithRankAtMost(input(input_idx), 1, &input_shape));
 
   requested_input_tensor_as_partial_shape_[input_idx] = true;
-  if (input_idx < input_tensors_as_shapes_.size() &&
+  int input_tensors_as_shapes_size_ = input_tensors_as_shapes_.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/shape_inference.cc:723]: `const` here and below, in all places we are extracting a variable

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Still need to `const` qualify temporaries

File: tensorflow/core/framework/shape_inference.cc [hunk 3]
```diff
@@ -739,8 +739,8 @@ Status InferenceContext::MakeShapeFromShapeTensor(int input_idx,
   TF_RETURN_IF_ERROR(WithRank(input(input_idx), 1, &input_shape));
 
   requested_input_tensor_as_partial_shape_[input_idx] = true;
-  int input_tensors_as_shapes_size_ = input_tensors_as_shapes_.size();
-  if (input_idx < input_tensors_as_shapes_size_ &&
+  int input_tensors_as_shapes_size = input_tensors_as_shapes_.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/shape_inference.cc:742]: `const`

File: tensorflow/core/framework/shape_inference.cc [hunk 4]
```diff
@@ -1099,14 +1103,16 @@ Status InferenceContext::AttachContext(const Status& status) {
   std::vector<string> input_from_tensors_str;
   std::vector<string> input_from_tensors_as_shape_str;
   input_from_tensors_as_shape_str.reserve(inputs_.size());
-  for (int i = 0; i < inputs_.size(); ++i) {
+  for (int i = 0, iter_limit = inputs_.size(); i < iter_limit; ++i) {
+    int input_tensors_size_ = input_tensors_.size();
+    int input_tensors_as_shapes_size_ = input_tensors_as_shapes_.size();
     if (requested_input_tensor_as_partial_shape_[i] &&
-        i < input_tensors_as_shapes_.size() &&
+        i < input_tensors_as_shapes_size_ &&
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/shape_inference.cc:1110]: This needs to be changed, variable name changed

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/util/bcast.h [hunk 5]
```diff
@@ -139,7 +139,8 @@ BCastList<N>::BCastList(const BCastList::Vec (&x)[N],
     if (x[i] != x[0]) {
       all_equal = false;
     }
-    if (static_cast<int>(x[i].size()) > largest_rank) {
+    int x_i_size = x[i].size();
```
Comment (mihaimaruseac) [inline tensorflow/core/util/bcast.h:142]: `const` since it's a local variable.

File: tensorflow/core/grappler/utils/topological_sort.cc [hunk 6]
```diff
@@ -81,8 +81,8 @@ Status ComputeTopologicalOrder(
     int ready_node = (*ready_nodes)[front];
     for (int fanout : graph_view.GetFanout(ready_node)) {
       ++num_ready_inputs[fanout];
-      if (num_ready_inputs[fanout] ==
-          static_cast<int>(graph_view.GetFanin(fanout).size())) {
+      int graph_view_GetFanin_fanout_size = graph_view.GetFanin(fanout).size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/utils/topological_sort.cc:84]: `const` since it's a local variable

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Still need to `const`-qualify local variables introduced by the change

File: tensorflow/core/grappler/utils/topological_sort.cc [hunk 7]
```diff
@@ -96,8 +96,8 @@ Status ComputeTopologicalOrder(
                  "at node = "
               << graph.node(back).DebugString();
       for (int i = 0; i < graph_view.num_nodes(); ++i) {
-        if (num_ready_inputs[i] !=
-            static_cast<int>(graph_view.GetFanin(i).size())) {
+        int graph_view_GetFanin_i_size = graph_view.GetFanin(i).size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/utils/topological_sort.cc:99]: This one too.

File: tensorflow/core/util/bcast.h [hunk 8]
```diff
@@ -176,7 +177,8 @@ BCastList<N>::BCastList(const BCastList::Vec (&x)[N],
 
   // 1-extend and align all vectors.
   for (int i = 0; i < N; ++i) {
-    if (static_cast<int>(copy[i].size()) < largest_rank) {
+    int copy_i_size = copy[i].size();
```
Comment (mihaimaruseac) [inline tensorflow/core/util/bcast.h:180]: This one too

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@tg-at-google Can you please check @mihaimaruseac's comments and keep us posted. Thanks!

**Comment by tg-at-google** [comment]
@gbaned updated.

File: tensorflow/core/framework/shape_inference.cc (hunk 1 above)
Comment (mihaimaruseac) [reply tensorflow/core/framework/shape_inference.cc:185]: Not `const`, still ending in `_`

File: tensorflow/core/framework/shape_inference.cc (hunk 2 above)
Comment (mihaimaruseac) [reply tensorflow/core/framework/shape_inference.cc:723]: Not `const`, still ending in `_`.

File: tensorflow/core/framework/shape_inference.cc [hunk 9]
```diff
@@ -736,7 +739,8 @@ Status InferenceContext::MakeShapeFromShapeTensor(int input_idx,
   TF_RETURN_IF_ERROR(WithRank(input(input_idx), 1, &input_shape));
 
   requested_input_tensor_as_partial_shape_[input_idx] = true;
-  if (input_idx < input_tensors_as_shapes_.size() &&
+  int input_tensors_as_shapes_size_ = input_tensors_as_shapes_.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/shape_inference.cc:742]: Should be `const` and not end with `_`

File: tensorflow/core/framework/shape_inference.cc [hunk 10]
```diff
@@ -1099,14 +1103,16 @@ Status InferenceContext::AttachContext(const Status& status) {
   std::vector<string> input_from_tensors_str;
   std::vector<string> input_from_tensors_as_shape_str;
   input_from_tensors_as_shape_str.reserve(inputs_.size());
-  for (int i = 0; i < inputs_.size(); ++i) {
+  for (int i = 0, end = inputs_.size(); i < end; ++i) {
+    int input_tensors_size_ = input_tensors_.size();
+    int input_tensors_as_shapes_size_ = input_tensors_as_shapes_.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/shape_inference.cc:1108]: Should be `const` and not end with `_`

File: tensorflow/core/framework/shape_inference.cc (hunk 4 above)
Comment (mihaimaruseac) [reply tensorflow/core/framework/shape_inference.cc:1110]: Still unchanged to not end in `_`

File: tensorflow/core/framework/shape_inference.cc [hunk 11]
```diff
@@ -1099,14 +1103,16 @@ Status InferenceContext::AttachContext(const Status& status) {
...
     if (requested_input_tensor_as_partial_shape_[i] &&
-        i < input_tensors_as_shapes_.size() &&
+        i < input_tensors_as_shapes_size_ &&
         input_tensors_as_shapes_[i].IsSet() &&
         RankKnown(input_tensors_as_shapes_[i])) {
       input_from_tensors_as_shape_str.push_back(strings::StrCat(
           "input[", i, "] = ", DebugString(input_tensors_as_shapes_[i])));
-    } else if (requested_input_tensor_[i] && i < input_tensors_.size() &&
+    } else if (requested_input_tensor_[i] && i < input_tensors_size_ &&
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/shape_inference.cc:1115]: Local variable ending in `_`

File: tensorflow/core/grappler/utils/topological_sort.cc [hunk 12]
```diff
@@ -81,8 +81,8 @@ Status ComputeTopologicalOrder(
     int ready_node = (*ready_nodes)[front];
     for (int fanout : graph_view.GetFanout(ready_node)) {
       ++num_ready_inputs[fanout];
-      if (num_ready_inputs[fanout] ==
-          static_cast<int>(graph_view.GetFanin(fanout).size())) {
+      const int graph_view_GetFanin_fanout_size = graph_view.GetFanin(fanout).size();
+      if (num_ready_inputs[fanout] == graph_view_GetFanin_fanout_size) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/utils/topological_sort.cc:85]: Local variable name should be `snake_case`, please remove capitalization in `GetFanin` substring

File: tensorflow/core/util/bcast.h [hunk 13]
```diff
@@ -139,7 +139,8 @@ BCastList<N>::BCastList(const BCastList::Vec (&x)[N],
     if (x[i] != x[0]) {
       all_equal = false;
     }
-    if (static_cast<int>(x[i].size()) > largest_rank) {
+    const int x_i_size = x[i].size();
+    if (x_i_size > largest_rank) {
```
Comment (mihaimaruseac) [inline tensorflow/core/util/bcast.h:143]: This is better as a cast

Comment (mihaimaruseac) [reply tensorflow/core/util/bcast.h:143]: Actually, change type of `largest_rank` in line 136, at declaration time.

File: tensorflow/core/util/bcast.h [hunk 14]
```diff
@@ -176,7 +177,8 @@ BCastList<N>::BCastList(const BCastList::Vec (&x)[N],
 
   // 1-extend and align all vectors.
   for (int i = 0; i < N; ++i) {
-    if (static_cast<int>(copy[i].size()) < largest_rank) {
+    const int copy_i_size = copy[i].size();
+    if (copy_i_size < largest_rank) {
```
Comment (mihaimaruseac) [inline tensorflow/core/util/bcast.h:181]: Should be handled better by changing type of `largest_rank` local variable

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Should have made a separate PR with the fixes from previous ones (e.g., replacing `iter_size` with `end` in for loops should have been in a different PR instead of being tacked on the currently opened ones)

File: tensorflow/core/util/bcast.h (hunk 14 above)
Comment (mihaimaruseac) [reply tensorflow/core/util/bcast.h:180]: Now that the type changed we no longer need the copy

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/utils/topological_sort.cc [hunk 15]
```diff
@@ -96,8 +96,8 @@ Status ComputeTopologicalOrder(
                  "at node = "
               << graph.node(back).DebugString();
       for (int i = 0; i < graph_view.num_nodes(); ++i) {
-        if (num_ready_inputs[i] !=
-            static_cast<int>(graph_view.GetFanin(i).size())) {
+        const int graph_view_GetFanin_i_size = graph_view.GetFanin(i).size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/utils/topological_sort.cc:99]: Variable name should be `snake_case`, not `snake_case_WithCamelCase_and_snake_case_again`

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/utils/topological_sort.cc [hunk 16]
```diff
@@ -96,8 +96,8 @@ Status ComputeTopologicalOrder(
                  "at node = "
               << graph.node(back).DebugString();
       for (int i = 0; i < graph_view.num_nodes(); ++i) {
-        if (num_ready_inputs[i] !=
-            static_cast<int>(graph_view.GetFanin(i).size())) {
+        const int graph_view_Get_fanin_i_size = graph_view.GetFanin(i).size();
+        if (num_ready_inputs[i] != graph_view_Get_fanin_i_size) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/utils/topological_sort.cc:100]: Still has a capital letter. Why not use the shorter name `max_size`? It's a local variable, defined just above, not used anywhere else, so it does not harm readability.

And then also change the other local above (line 84) to be `max_size` too.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/utils/topological_sort.cc (hunk 16 above)
Comment (tg-at-google) [reply tensorflow/core/grappler/utils/topological_sort.cc]: Made the change.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by tg-at-google** [comment]
build failed due to undeclared, `copy_i_size` in `bcast.h`, updated `bcast.h` to resolve.

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**